aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,221 @@
+    private static final List<String> privilegeList = Arrays.asList("SELECT", "INSERT", "DELETE");
+
+    LakeFormationCredentialsProvider(Map<String, String> properties) {
+      AwsSessionCredentials awsSessionCredentials = getAssumeRoleCredentials(properties);
+      String region = properties.containsKey(AwsProperties.LAKE_FORMATION_AWS_REGION) ?
+          properties.get(AwsProperties.LAKE_FORMATION_AWS_REGION) : null;
+
+      lakeFormationClient = buildLakeFormationClient(awsSessionCredentials.accessKeyId(),
+          awsSessionCredentials.secretAccessKey(), awsSessionCredentials.sessionToken(), region);
+      tableArn = getTableArn(properties);
+    }
+
+    @Override
+    public AwsCredentials resolveCredentials() {
+      GetTemporaryGlueTableCredentialsRequest getTemporaryGlueTableCredentialsRequest =
+          GetTemporaryGlueTableCredentialsRequest.builder()
+              .tableArn(tableArn)
+              .permissionsWithStrings(privilegeList)
+              .build();
+      GetTemporaryGlueTableCredentialsResponse response =
+          lakeFormationClient.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
+      return AwsSessionCredentials.create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+    }
+
+    private AwsSessionCredentials getAssumeRoleCredentials(Map<String, String> properties) {
+      String roleArn = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_ARN);
+      Preconditions.checkNotNull(roleArn, "Cannot initialize LakeFormationAwsClientFactory with empty roleArn");
+      String sessionName = String.format("iceberg-aws-%s", UUID.randomUUID());
+      String externalId = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_EXTERNAL_ID);
+      String tagString = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_TAGS);
+      Preconditions.checkNotNull(tagString, "Cannot initialize LakeFormationAwsClientFactory with empty tags");
+      List<String> tokens = Splitter.on(',').splitToList(tagString);
+      String key = tokens.get(0).substring("Key=".length());
+      String value = tokens.get(1).substring("Value=".length());
+      Tag tag = Tag.builder().key(key).value(value).build();
+      int timeout = PropertyUtil.propertyAsInt(properties,
+          AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC, AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC_DEFAULT);
+      AssumeRoleRequest request = AssumeRoleRequest.builder()
+          .roleArn(roleArn)
+          .roleSessionName(sessionName)
+          .durationSeconds(timeout)
+          .externalId(externalId)
+          .tags(tag)
+          .build();
+      return (AwsSessionCredentials) StsAssumeRoleCredentialsProvider.builder()
+          .stsClient(StsClient.builder().httpClientBuilder(UrlConnectionHttpClient.builder()).build())
+          .refreshRequest(request)
+          .build().resolveCredentials();
+    }
+
+    private LakeFormationClient buildLakeFormationClient(String accessKey,
```

Comment:
we don't need to build LF client again here, we can update the constructor to take `LakeFormationClient` as input, and pass it in from the client factory.

Reply:
I see, you are right, the LF client is more like the STS client that is used to simply generate a credential, so it does not need to be shared across all factories.